scripts/validate.py

```diff
@@ -43,81 +68,110 @@ def validate_item_targeting(item, for_exp=False):
         try:
             result = list(EVALUATOR.validate(jexl_expression))
             if len(result) > 0:
-                raise Exception(result[0])
-        except Exception as e:
+                raise SyntaxError(result[0])
+        except SyntaxError as e:
             print(e)
             sys.exit(1)
 
 
-def load_all_schemas():
-    if len(ALL_SCHEMAS):
-        return
+def validate_action(action, for_exp):
+    indentation = "\t" if for_exp else ""
+    all_action_schemas = load_schema("action")
+    action_type = action["type"]
+
+    print("{}Validate message action {}".format(indentation, action_type))
+    if action_type not in all_action_schemas:
+        print("{}Unknown action {}".format(indentation, action_type))
+        sys.exit(1)
+
+    schema = all_action_schemas[action_type]
+    try:
+        jsonschema.validate(instance=action, schema=schema)
+    except ValidationError as err:
+        match = best_match([err])
+        print("{}Validation error: {}".format(indentation, match.message))
+        sys.exit(1)
+
 
-    possible_schemas = [
-        "schema/cfr.schema.json",
-        "schema/onboarding.schema.json",
-        "schema/cfr-fxa.schema.json",
-        "schema/cfr-heartbeat.schema.json",
-        "schema/messaging-experiments.schema.json",
-        "schema/whats-new-panel.schema.json",
-    ]
-    for path in possible_schemas:
-        with open(path, "r") as f:
-            ALL_SCHEMAS[path] = json.loads(f.read())
+def extract_actions(message, message_type):
+    def _extract_cfr():
+        for name, button in message["content"].get("buttons").items():
```

Comment:
We mix moments and cfr in the same file so this will need to be guarded somehow.

Reply:
Good point.